cfme/infrastructure/host.py

```diff
@@ -92,15 +92,29 @@ def __init__(self, **kwargs):
             super(Host.Credential, self).__init__(**kwargs)
             self.ipmi = kwargs.get('ipmi')
 
-    def update(self, updates, validate_credentials=False):
+    def update(self, updates, validate_credentials=False, action='edit_from_details'):
```

Comment:
I took a look at varmeth. We can discuss further about updating in a nother PR, but at first glance I am not sure varmeth is the correct solution. I might could be convinced otherwise, but the pieces I would use to build the variable methods are all similar with just specific pieces added or removed. Coding for each context would create a bunch of duplicate code. Seems like adding conditionals via kwargs (or another solution) might work better.  IMHO

Reply:
Yup, just a suggestion. It seems like it'd be useful for things like this, but I haven't tried it myself yet :smile: